includes/class-gam.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * Newspack Ads GAM Integration.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack_Network;
+
+use Newspack_Ads\Providers\GAM\Api as GAM_API;
+use Newspack_Ads\Providers\GAM_Model;
+
+/**
+ * Integration class for Newspack Ads' GAM support.
+ */
+final class GAM {
+	/**
+	 * Initialize hooks.
+	 */
+	public static function init() {
+		add_action( 'newspack_ads_setup_gam', [ __CLASS__, 'create_targeting_keys' ] );
+		add_action( 'save_post_' . Hub\Nodes::POST_TYPE_SLUG, [ __CLASS__, 'create_targeting_keys' ] );
+		add_filter( 'newspack_ads_ad_targeting', [ __CLASS__, 'add_targeting' ], 10, 2 );
+	}
+
+	/**
+	 * Create targeting keys.
+	 */
+	public static function create_targeting_keys() {
+		if ( ! Site_Role::is_hub() ) {
+			return;
+		}
+		if ( ! class_exists( 'Newspack_Ads\Providers\GAM_Model' ) ) {
+			return;
+		}
+		$api = GAM_Model::get_api();
```

Comment:
Don't we need a check or try-catch here to avoid errors if the Ads plugin is not active?

Reply:
This method never throws an exception. It returns false if the API is not available.

Reply:
oh I missed the `class_exists` check right there on my face